auto_nag/scripts/topcrash_add_keyword.py

```diff
@@ -0,0 +1,165 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+
+import re
+
+from libmozdata.bugzilla import Bugzilla
+from libmozdata.connection import Connection
+
+from auto_nag import utils
+from auto_nag.bzcleaner import BzCleaner
+from auto_nag.topcrash import Topcrash
+
+# TODO: should be moved when resolving https://github.com/mozilla/relman-auto-nag/issues/1384
+HIGH_SEVERITY = {"S1", "critical", "S2", "major"}
+
+MAX_SIGNATURES_PER_QUERY = 30
+
+
+class TopcrashAddKeyword(BzCleaner):
+    def description(self):
+        return "Bugs with missing topcrash keywords"
+
+    def columns(self):
+        return ["id", "summary", "severity", "added_keyword"]
+
+    def handle_bug(self, bug, data):
+        bugid = str(bug["id"])
+        if bugid in data:
+            return
+
+        has_topcrash_keyword = "topcrash" in bug["keywords"]
+        has_startup_keyword = "topcrash-startup" in bug["keywords"]
+        if has_topcrash_keyword and has_startup_keyword:
+            return
+
+        signatures = utils.get_signatures(bug["cf_crash_signature"])
+        has_topcrash_signature = any(
+            signature in self.topcrashes for signature in signatures
+        )
+        assert has_topcrash_signature
+
+        has_startup_signature = any(
+            self.topcrashes.get(signature) for signature in signatures
+        )
```

Comment:
The data format of self.topcrashes is not so obvious: a dict from signature to whether the signature is a startup crash or not.
Might be clearer to have two separate sets, a set of top signatures and a set of top startup signatures (some signatures will be in both sets).
Or a dict from signature name to dicts with `is_startup` property:
```Python
{
    "signature1": {
        "is_startup": False,
    },
    ...
}
```

Reply:
Thank you! I agree with you, this will serve beyond just readability. It will improve the extensibility since we can add more info like ranks, volumes or anything we may need later.

Done in 268e84d5ecd696a5eb591f24b579cf3c658eeb98.